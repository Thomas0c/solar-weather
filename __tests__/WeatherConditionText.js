import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import WeatherConditionText from '../lib/js/app/components/weatherConditionText';

test('renders correctly when showDetails is true', () => {
  const tree = renderer.create(<WeatherConditionText text="Test Text" showDetails />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly when showDetails is false', () => {
  const tree = renderer.create(<WeatherConditionText text="Test Text" showDetails={false} />).toJSON();
  expect(tree).toMatchSnapshot();
});
