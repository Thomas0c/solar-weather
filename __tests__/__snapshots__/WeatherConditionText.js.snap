// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when showDetails is false 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityLiveRegion={undefined}
  accessibilityTraits={undefined}
  accessibilityViewIsModal={undefined}
  accessible={undefined}
  collapsable={undefined}
  hitSlop={undefined}
  importantForAccessibility={undefined}
  needsOffscreenAlphaCompositing={undefined}
  onAccessibilityTap={undefined}
  onLayout={undefined}
  onMagicTap={undefined}
  pointerEvents={undefined}
  removeClippedSubviews={undefined}
  renderToHardwareTextureAndroid={undefined}
  shouldRasterizeIOS={undefined}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
>
  <Text
    accessible={true}
    adjustsFontSizeToFit={undefined}
    allowFontScaling={true}
    ellipsizeMode="tail"
    minimumFontScale={undefined}
    numberOfLines={undefined}
    onLayout={undefined}
    onLongPress={undefined}
    onPress={undefined}
    pressRetentionOffset={undefined}
    selectable={undefined}
    selectionColor={undefined}
    style={
      Object {
        "color": "#EFEFEF",
        "lineHeight": 25,
      }
    }
    suppressHighlighting={undefined}
    testID={undefined}
    textBreakStrategy={undefined}
  >
    Test Text
  </Text>
</View>
`;

exports[`renders correctly when showDetails is true 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityLiveRegion={undefined}
  accessibilityTraits={undefined}
  accessibilityViewIsModal={undefined}
  accessible={undefined}
  collapsable={undefined}
  hitSlop={undefined}
  importantForAccessibility={undefined}
  needsOffscreenAlphaCompositing={undefined}
  onAccessibilityTap={undefined}
  onLayout={undefined}
  onMagicTap={undefined}
  pointerEvents={undefined}
  removeClippedSubviews={undefined}
  renderToHardwareTextureAndroid={undefined}
  shouldRasterizeIOS={undefined}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
>
  <Text
    accessible={true}
    adjustsFontSizeToFit={undefined}
    allowFontScaling={true}
    ellipsizeMode="tail"
    minimumFontScale={undefined}
    numberOfLines={undefined}
    onLayout={undefined}
    onLongPress={undefined}
    onPress={undefined}
    pressRetentionOffset={undefined}
    selectable={undefined}
    selectionColor={undefined}
    style={
      Object {
        "color": "#EFEFEF",
        "lineHeight": 25,
      }
    }
    suppressHighlighting={undefined}
    testID={undefined}
    textBreakStrategy={undefined}
  >
    Test Text
  </Text>
</View>
`;
